src/AudioAnalysisTools/LongDurationSpectrograms/LDSpectrogramRGB.cs

```diff
@@ -540,10 +545,38 @@ public void DrawGreyScaleSpectrograms(DirectoryInfo opdir, string opFileName, st
                     continue;
                 }
 
-                // the directory for the following path must exist
-                var path = FilenameHelpers.AnalysisResultPath(opdir, opFileName, key, "png");
                 var bmp = this.DrawGreyscaleSpectrogramOfIndex(key);
-                bmp?.Save(path);
+
+                var header = new Bitmap(bmp.Width, 20);
+                Graphics g = Graphics.FromImage(header);
+                g.Clear(Color.LightGray);
+                g.SmoothingMode = SmoothingMode.AntiAlias;
+                //g.InterpolationMode = InterpolationMode.HighQualityBicubic;
+                //g.PixelOffsetMode = PixelOffsetMode.HighQuality;
+                g.DrawString(key, new Font("Tahoma", 9), Brushes.Black, 4, 4);
+                var indexImage = ImageTools.CombineImagesVertically(new List<Image>(new Image[] { header, bmp }));
+
+                // save the image - the directory for the path must exist
+                //var path = FilenameHelpers.AnalysisResultPath(opdir, opFileName, key, "png");
+                //indexImage?.Save(path);
+
+                listOfImages.Add(indexImage);
+            }
+
+            // save the combined track image - the directory for the path must exist
+            var path = FilenameHelpers.AnalysisResultPath(opdir, opFileName, "KEYS", "png");
```

Comment:
This change breaks existing functionality. Output each image, separately, please.

If you want to output a combined image, it needs to hidden behind a config setting that is not enabled by default.